core/src/main/scala/org/apache/spark/sql/execution/columnar/impl/ColumnFormatRelation.scala

```diff
@@ -417,8 +422,90 @@ class ColumnFormatRelation(
     })
     ColumnFormatRelation.flushLocalBuckets(resolvedName)
   }
+
+  /**
+    * Index table is same as the column table apart from how it is
+    * partitioned and colocated. Add GEM_PARTITION_BY and GEM_COLOCATE_WITH
+    * clause in its options. Also add GEM_INDEXED_TABLE parameter to
+    * indicate that this is an index table.
+    */
+  private def createIndexTable(indexIdent: QualifiedTableName,
+                               tableIdent: QualifiedTableName,
+                               tableRelation: JDBCAppendableRelation,
+                               indexColumns: Seq[String],
+                               options: Map[String, String]): Unit = {
+
+
+    val parameters = new CaseInsensitiveMutableHashMap(options)
+    val snc = _context.asInstanceOf[SnappyContext]
+    val indexTblName = snc.getIndexTable(indexIdent).toString
+    val caseInsensitiveMap = new CaseInsensitiveMutableHashMap(tableRelation.origOptions)
+    val tempOptions = caseInsensitiveMap.filterNot(pair => {
+      pair._1.equals(Utils.toLowerCase(StoreUtils.PARTITION_BY)) ||
+        pair._1.equals(Utils.toLowerCase(StoreUtils.COLOCATE_WITH)) ||
+        pair._1.equals(Utils.toLowerCase(JdbcExtendedUtils.DBTABLE_PROPERTY)) ||
+        pair._1.equals(Utils.toLowerCase(ExternalStoreUtils.INDEX_NAME))
+    }).toMap + (StoreUtils.PARTITION_BY -> indexColumns.mkString(",")) +
+      (StoreUtils.GEM_INDEXED_TABLE -> tableIdent.toString) +
+      (JdbcExtendedUtils.DBTABLE_PROPERTY -> indexTblName)
+
```

Comment:
For row tables, see the full CREATE INDEX syntax: https://db.apache.org/derby/docs/10.4/ref/rrefsqlj20937.html
Notice the ASC/DESC specifications for individual columns. We need that for column indexes too (at least going forward). The proper way in API to accept that will be to use the "Column" class (optionally) like in DataFrame. See the DataFrame.sort() overloads and lets match that.